src/main/java/org/opensearch/sdk/handlers/ExtensionsInitRequestHandler.java

```diff
@@ -48,6 +48,7 @@ public ExtensionsInitRequestHandler(ExtensionsRunner extensionsRunner) {
     public InitializeExtensionResponse handleExtensionInitRequest(InitializeExtensionRequest extensionInitRequest) {
         logger.info("Registering Extension Request received from OpenSearch");
         extensionsRunner.opensearchNode = extensionInitRequest.getSourceNode();
+        extensionsRunner.getThreadPool().getThreadContext().putHeader("extension_unique_id", extensionInitRequest.getExtension().getId());
```

Comment:
Trying to understand, the header is plumbed in the threadcontext how does security plugin on OpenSearch see this?

Reply:
If I understand the logic in the security plugin correctly, a node is identified as being a node in the cluster iff one of 2 conditions are met:

1. The node's principal (extracted from the node cert) matches a node in the static nodes_dn list from `plugins.security.nodes_dn` or from the dynamic list in `nodes_dn.yml` and dynamic nodes enabled
2. The OID extracted from the certificates subjectAltName matches the `plugins.security.cert.oid` setting which defaults to `1.2.3.4.5.5` (Curious, why is this chosen as a default?)

In addition to one of the 2 methods up above, the security plugin also expects the `_opendistro_security_remotecn` thread context header is set. If this header on a transport request matches the cluster name from the cluster state then its identified as a transport request originating from another node in the cluster. If it doesn't match, then its a trusted cross cluster request. 

`extension_unique_id` here mimics the `_opendistro_security_remotecn` behavior.

Reply:
Thank you, that makes more sense.

Reply:
I've started making changes to move the identity into abstract piece of information. This should help us add more information for identity for extension and it encapsulates future changes.
See: https://github.com/opensearch-project/OpenSearch/pull/7302/files#diff-10ffb20227a1e7e024bc7a633c75b97ea7d2db8e59b6117de39e28b7b9e4c22a

Reply:
Thank you Sarat! wdyt about populating the threadcontext with the unique id from that protobuf when its received in core? Similar to how `_opendistro_security_remotecn` is expected to be in the threadcontext, or would there be a better place to keep track of where the transport request originated from that can be read by other plugins?

Reply:
Lets do it, Lets start with populating `uniqueId` from ExtensionIdentity.
I think this is the only way it would work, considering if Extension wants to talk to a plugin the threadcontext is expected.